elastic/eui
# Create new EuiTextBlockTruncate component

## Summary

> ⚠️ I suggest [turning off whitespace changes](https://github.com/elastic/eui/pull/7250/files?diff=split&w=1) when reviewing diffs, as docs changes contain a good amount of "changes" that are just indentation

closes #6312

Part 1 of #7226 (note: does not close the issue, as @walterra would like the component to be baked into EuiBasicTable's `columns` API - I'm separating that second part into another PR for easier review)

![screencap](https://github.com/elastic/eui/assets/549407/aae0363b-90b8-4f33-a6b3-f71a55095654)

## QA

- https://eui.elastic.co/pr_7250/#/utilities/text-truncation/multi-line
- Go to https://eui.elastic.co/pr_7250/storybook?path=/story/euitextblocktruncate--playground
- [x] Confirm the `lines` truncates as expected
- Toggle the `cloneElement` control to true and inspect the text in the DOM. Confirm the parent element is now a `<p>` tag and not a `<div>` (toggle back and forth to see the difference)

### General checklist

- Browser QA
    ~- [ ] Checked in both **light and dark** modes~
- Docs site QA
    ~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A, designers aren't really sure how to add truncation to Figma
